bin/lint-migrations-file/src/change_set/strict.clj

```diff
@@ -122,6 +122,30 @@
    (some change-types-supporting-rollback (mapcat keys changes))
    (contains? change-set :rollback)))
 
+(def ^:private change-types-requiring-preconditions
+  #{:createTable
+    :dropTable
+    :addColumn
+    :dropColumn
+    :createIndex
+    :dropIndex
+    :addForeignKeyConstraint
+    :dropForeignKeyConstraint})
+
+(defn- precondition-present-when-required?
+  "Ensures that certain changeSet types include a preCondition. The intent is for the preCondition to ensure that the changeSet is
+  idempotent by checking whether the table/column/index/etc does not exist before trying to create it. (Or inversely, that it does
+  exist before trying to drop it.)
+
+  We don't currently assert on the structure of the preCondition to provide flexibility if there are cases where idempotence is not
+  desired."
+  [{:keys [id changes] :as change-set}]
+  (or
+   (int? id)
+   (< (major-version id) 51)
+   (not-any? change-types-requiring-preconditions (mapcat keys changes))
+   (contains? change-set :preConditions)))
```

Comment:
should we check that the content is not empty too?

Reply:
I intentionally didn't do that so that you could leave a blank `:preConditions` field on a changeset, similar to with rollbacks. I figured it's plausible there could be a case where this isn't desired, so I don't want to make it too rigid—at least to start. Maybe we can see how this goes and make it stricter if it seems like it doesn't cause issues.